### DISCUSSION TITLE
Connecticut mukeys do not match: WCS <> TIF <> SDA

### DISCUSSION THREAD

**Discussion by kevinwolz** [ncss-tech/soilDB#334] [Q&A] [answered]
I'm experience a very weird situation that is seemingly specific and unique to Connecticut. To illustrate, let's use Litchfield County, CT:

```
aoi <- tigris::counties(state = "CT", cb = TRUE) %>%
  dplyr::filter(NAME == "Litchfield")
```

I can get all mukeys in this county (plus some stragglers in adjacent MA and NY to round out the rectangular raster) in three different ways:
```
mukey_clean <- function(x) as.numeric(unique(x)[,1])

mukeys.wcs <- soilDB::mukey.wcs(aoi = aoi, db = "gssurgo", res = 30, quiet = TRUE) %>% 
  mukey_clean()

mukeys.tif <- "FY2023_gSSURGO-mukey.tif" %>%
  terra::rast() %>%
  terra::crop(terra::project(terra::vect(aoi), terra::crs(.)), snap = "out") %>%
  mukey_clean()

mukeys.SDA <- soilDB::SDA_spatialQuery(aoi, what = "mukey", db = "SSURGO") %>%
  dplyr::pull(mukey) %>%
  unique()
```

`mukey.wcs` and `mukey.tif` generally match, with most mukeys being in the 200,000s. For example:
`285959  285962  285963  285965  285968`

`mukeys.SDA`, however, is totally different, with most mukeys being in the 3,000,000s. For example:
`3395767 3395769 3395770 3395771 3395772`

When I then retrieve basic tabular data for these mukeys from SDA, I get very different results. Neither `mapunit.wcs` nor `mapunit.tif` contain any data from CT survey areas, but `mapunit.SDA` does. 
```
make_where <- function(mukeys) {
  mukeys %>%
    as.integer() %>%
    paste0("'", ., "'") %>%
    toString() %>%
    paste0("mukey IN (", ., ")")
}

mapunit.wcs <- soilDB::get_mapunit_from_SDA(WHERE = make_where(mukeys.wcs))
mapunit.tif <- soilDB::get_mapunit_from_SDA(WHERE = make_where(mukeys.tif))
mapunit.SDA <- soilDB::get_mapunit_from_SDA(WHERE = make_where(mukeys.SDA))

table(mapunit.wcs$areasymbol)
# MA003 MA607 MA608 NY021 NY027 NY079 
# 92    72    38    37    94    57

table(mapunit.tif$areasymbol)
# MA003 MA608 NY027 NY079 
# 87    11    86    34 

table(mapunit.SDA$areasymbol)
# CT601 MA003 MA608 NY027 
# 222    17     6    10 
```

Did Connecticut revamp its mukey system recently? Is that what I'm seeing here? 
Is the WCS and TIF on the old system and SDA on the new system?
Do you think this will be rectified once the WCS and TIF are updated to FY24?

**Comment by brownag** [comment, marked as answer]
> Did Connecticut revamp its mukey system recently? Is that what I'm seeing here? 

CT was split into an eastern part and western part in FY24 release; CT600 -> CT601 and CT602. This means all new MUKEYs.

> Is the WCS and TIF on the old system and SDA on the new system? 

Yes. SoilWeb WCS is still using 2023 data, and Box does not have the 2024 grids yet. SDA is updated on October 1, start of fiscal year

> Do you think this will be rectified once the WCS and TIF are updated to FY24?

Yes

**Reply by kevinwolz** [reply]
Great, thank you! I'm glad on was on the right track.

Congratulations, Connecticut!
